### PR TITLE
Fix GH-995: Fix Redirection Problem

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

Fixes redirection problem after logging onto Mattermost to Focalboard

#### Ticket Link

Fixes GH-995

Requires
https://github.com/mattermost/mattermost-webapp/pull/8632

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -87,13 +87,19 @@ const App = React.memo((): JSX.Element => {
...
                                     path='/workspace/:workspaceId/:boardId?/:viewId?'
-                                    render={({match}) => {
+                                    render={({match, location}) => {
                                         if (loggedIn === false) {
-                                            let redirectUrl = '/' + Utils.buildURL(`/workspace/${match.params.workspaceId}/`)
+                                            let redirectUrl = '/' + Utils.buildURL(`/workspace/${match.params.workspaceId}${match.params.boardId ? `/${match.params.boardId}` : ''}${match.params.viewId ? `/${match.params.viewId}` : ''}${location.search}`)
                                             if (redirectUrl.indexOf('//') === 0) {
                                                 redirectUrl = redirectUrl.slice(1)
                                             }
-                                            const loginUrl = `/login?r=${encodeURIComponent(redirectUrl)}`
```
Comment (chenilim) [inline webapp/src/app.tsx:96]: This URL should be used when not running as a plugin, so Personal Server works.

**Review by chenilim — CHANGES_REQUESTED** [review]
Need to change the URL for non-plugin as we discussed. Otherwise this LGTM.

File: webapp/src/app.tsx (hunk 1 above)
Comment (jespino) [reply webapp/src/app.tsx:96]: Actually, this url is the same for both, it is the same url for mattermost-webapp, so I think it is fine.

**Review by jespino — APPROVED** [review]
LGTM.

**Comment by hahmadia** [comment]
Not needed anymore due to Miguels changes on Mattermost web app code

**Pull request closed without merging**